Intermittent Python test failure in test_audio_functionality.py

Repro steps:
```
pytest test/test_audio_functionality.py
```

Expected:
Passes consistently

Actual:
Fails intermittently, see the output here:
```
➜  turicreate git:(master) pytest test/test_audio_functionality.py 
=========================================================================================== test session starts ============================================================================================
platform darwin -- Python 2.7.10, pytest-3.2.5, py-1.8.0, pluggy-0.4.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/zach/turicreate2/debug/src/unity/python/turicreate/.hypothesis/examples')
rootdir: /Users/zach/turicreate2/debug/src/unity/python, inifile:
plugins: hypothesis-3.76.0
collected 32 items                                                                                                                                                                                          

test/test_audio_functionality.py ........s.......s.......s......F

================================================================================================= FAILURES =================================================================================================
____________________________________________________________________________________ ReuseDeepFeatures.test_simple_case ____________________________________________________________________________________

self = <turicreate.test.test_audio_functionality.ReuseDeepFeatures testMethod=test_simple_case>

    def test_simple_case(self):
        data = copy(binary_test_data)
        deep_features = tc.sound_classifier.get_deep_features(data['audio'])
    
        # Verify deep features in correct format
        self.assertTrue(isinstance(deep_features, tc.SArray))
        self.assertEqual(len(data), len(deep_features))
        self.assertEqual(deep_features.dtype, list)
        self.assertEqual(len(deep_features[0]), 3)
        self.assertTrue(isinstance(deep_features[0][0], np.ndarray))
        self.assertEqual(deep_features[0][0].dtype, np.float64)
        self.assertEqual(len(deep_features[0][0]), 12288)
    
        # Test helper methods
        self.assertTrue(tc.sound_classifier._is_audio_data_sarray(data['audio']))
        self.assertTrue(tc.sound_classifier._is_deep_feature_sarray(deep_features))
    
        original_audio_data = data['audio']
        del data['audio']
    
        # Create a model using the deep features
        data['features'] = deep_features
        model = tc.sound_classifier.create(data, 'labels', feature='features')
    
        # Test predict
        predictions_from_audio = model.predict(original_audio_data, output_type='probability_vector')
        predictions_from_deep_features = model.predict(deep_features, output_type='probability_vector')
        for a, b in zip(predictions_from_audio, predictions_from_deep_features):
>           self.assertAlmostEqual(list(a), list(b))
E           TypeError: unsupported operand type(s) for -: 'list' and 'list'

test/test_audio_functionality.py:398: TypeError
------------------------------------------------------------------------------------------- Captured stdout call -------------------------------------------------------------------------------------------

Training a custom neural network -
+------------------------+------------------------+------------------------+
| Epoch                  | Training Accuracy (%)  | Elapsed Time (seconds) |
+------------------------+------------------------+------------------------+
| 0                      | 1.000                  | 0.955                  |
+------------------------+------------------------+------------------------+
| 1                      | 1.000                  | 0.962                  |
+------------------------+------------------------+------------------------+
| 2                      | 1.000                  | 0.969                  |
+------------------------+------------------------+------------------------+
| 3                      | 1.000                  | 0.975                  |
+------------------------+------------------------+------------------------+
| 4                      | 1.000                  | 0.982                  |
+------------------------+------------------------+------------------------+
| 5                      | 1.000                  | 0.989                  |
+------------------------+------------------------+------------------------+
| 6                      | 1.000                  | 0.995                  |
+------------------------+------------------------+------------------------+
| 7                      | 1.000                  | 1.002                  |
+------------------------+------------------------+------------------------+
| 8                      | 1.000                  | 1.008                  |
+------------------------+------------------------+------------------------+
| 9                      | 1.000                  | 1.015                  |
+------------------------+------------------------+------------------------+
============================================================================= 1 failed, 28 passed, 3 skipped in 207.94 seconds =============================================================================
➜  turicreate git:(master) pytest test/test_audio_functionality.py
=========================================================================================== test session starts ============================================================================================
platform darwin -- Python 2.7.10, pytest-3.2.5, py-1.8.0, pluggy-0.4.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/zach/turicreate2/debug/src/unity/python/turicreate/.hypothesis/examples')
rootdir: /Users/zach/turicreate2/debug/src/unity/python, inifile:
plugins: hypothesis-3.76.0
collected 32 items                                                                                                                                                                                          

test/test_audio_functionality.py ........s.......s.......s.......

================================================================================== 29 passed, 3 skipped in 288.04 seconds ==================================================================================
➜  turicreate git:(master)
```

--- a/src/unity/python/turicreate/test/test_audio_functionality.py
+++ b/src/unity/python/turicreate/test/test_audio_functionality.py
@@ -445,19 +445,22 @@ class ReuseDeepFeatures(unittest.TestCase):
         predictions_from_audio = model.predict(original_audio_data, output_type='probability_vector')
         predictions_from_deep_features = model.predict(deep_features, output_type='probability_vector')
         for a, b in zip(predictions_from_audio, predictions_from_deep_features):
-            self.assertAlmostEqual(list(a), list(b))
+            np.testing.assert_array_almost_equal(a, b, decimal=6)
 
         # Test classify
         predictions_from_audio = model.classify(original_audio_data)
         predictions_from_deep_features = model.classify(deep_features)
         for a, b in zip(predictions_from_audio, predictions_from_deep_features):
-            self.assertAlmostEqual(a, b)
+            self.assertEqual(a['class'], b['class'])
+            np.testing.assert_array_almost_equal(a['probability'], b['probability'], decimal=6)
 
         # Test predict_topk
         predictions_from_audio = model.predict_topk(original_audio_data, k=2)
         predictions_from_deep_features = model.predict_topk(deep_features, k=2)
         for a, b in zip(predictions_from_audio, predictions_from_deep_features):
-            self.assertAlmostEqual(a, b)
+            self.assertEqual(a['id'], b['id'])
+            self.assertEqual(a['class'], b['class'])
+            np.testing.assert_array_almost_equal(a['probability'], b['probability'], decimal=6)
 
         # Test evaluate
         predictions_from_audio = model.evaluate(tc.SFrame({'features': original_audio_data,

